### PR TITLE
tmuxp: fix shell completion generation

### PR DESCRIPTION
**Pull request by otavio**
###### Description of changes

tmuxp 1.17+ uses shtab to generate the shell completion. This commit adapts the code to properly generate the shell completion module.

A side effect of use of shtab is that fish is not supported so it has been removed from postInstall.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/tmuxp/default.nix [hunk 1]
```diff
@@ -16,7 +16,10 @@ pypkgs.buildPythonApplication rec {
   # No tests in archive
   doCheck = false;
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [
+    pypkgs.shtab
```
Comment (AndersonTorres) [inline pkgs/tools/misc/tmuxp/default.nix:20]: I suggest to remove this pypkgs "macro"

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/tmuxp/default.nix (hunk 1 above)
Comment (otavio) [reply pkgs/tools/misc/tmuxp/default.nix]: Fixed.

**Review by marsam — APPROVED** [review]

**Comment by AndersonTorres** [comment]
fix the conflicts

**Comment by AndersonTorres** [comment]
Removing the approval label since the code became unmergeable afterwards.

**Comment by otavio** [comment]
@AndersonTorres the PR is now rebased.

**Pull request merged**